src/test/common/fdc3.app-channels.ts

```diff
@@ -137,11 +137,11 @@ export function createAppChannelTests(cc: ChannelControl<any,any>, documentation
       it(acTestId8, async () => {
         const errorMessage = `\r\nSteps to reproduce:\r\n- App A retrieves an app channel\r\n- App A switches to a different app channel\r\n- App A adds a context listener of type fdc3.instrument\r\n- App B retrieves the first channel that A retrieved\r\n- App B broadcasts a context of type fdc3.instrument${documentation}`;
 
-        let testChannel = await cc.createTestChannel()
+        const testChannel = await cc.createTestChannel()
         const resolveExecutionCompleteListener = cc.initCompleteListener(acTestId8)
-        testChannel = await cc.createTestChannel("a-different-test-channel");
-        cc.setupAndValidateListener1(testChannel, "unexpected-context", errorMessage, () => { /*noop*/ })
-        cc.openChannelApp(acTestId8, undefined, APP_CHANNEL_AND_BROADCAST)
+        const differentTestChannel = await cc.createTestChannel();
+        await cc.setupAndValidateListener1(testChannel, "unexpected-context", errorMessage, () => { /*noop*/ })
```

Comment:
THis is not positively testing that the context was retrieved on the testChannel and is not fully implementing the test. It needs to check that the context was received on testChannel and nothing was received on differentTestChannel